src/cocotb/share/lib/simulator/simulatormodule.cpp

```diff
@@ -538,6 +546,102 @@ static PyObject *register_value_change_callback(
     return rv;
 }
 
+// Register edge_count callback. Arguments:
+// 1) the signal handle, 2) the function to call, 3) the edge (int),
+// and 4) the count of edges to skip before the callback.
+// Remaining arguments and keyword arguments are to be passed to the callback
+static PyObject *register_edge_count_callback(
+    PyObject *, PyObject *args)  //, PyObject *keywds)
+{
+    if (!gpi_has_registered_impl()) {
+        PyErr_SetString(PyExc_RuntimeError, "No simulator available!");
+        return NULL;
+    }
+
+    Py_ssize_t numargs = PyTuple_Size(args);
+
+    if (numargs < 4) {
+        PyErr_SetString(PyExc_TypeError,
+                        "Attempt to register edge count callback without "
+                        "enough arguments!");
+        return NULL;
+    }
+
+    PyObject *pSigHdl = PyTuple_GetItem(args, 0);
+    if (Py_TYPE(pSigHdl) != &gpi_hdl_Object<gpi_sim_hdl>::py_type) {
+        PyErr_SetString(PyExc_TypeError,
+                        "First argument must be a gpi_sim_hdl");
+        return NULL;
+    }
+    gpi_sim_hdl sig_hdl = ((gpi_hdl_Object<gpi_sim_hdl> *)pSigHdl)->hdl;
+
+    // Extract the callback function
+    PyObject *function = PyTuple_GetItem(args, 1);
+    if (!PyCallable_Check(function)) {
+        PyErr_SetString(PyExc_TypeError,
+                        "Attempt to register edge count callback without "
+                        "passing a callable callback!");
+        return NULL;
+    }
+
+    long edge = PyLong_AsLong(PyTuple_GetItem(args, 2));
+    if ((edge == -1) && PyErr_Occurred()) {
+        return NULL;
+    }
+    if ((edge != GPI_RISING) && (edge != GPI_FALLING) &&
+        (edge != (GPI_RISING | GPI_FALLING))) {
+        PyErr_SetString(PyExc_TypeError,
+                        "Attempt to register value change callback for an "
+                        "invalid edge!");
+        return NULL;
+    }
+    unsigned long long count =
+        PyLong_AsUnsignedLongLong(PyTuple_GetItem(args, 3));
+    if ((count == (unsigned long long)-1) && PyErr_Occurred()) {
+        return NULL;
+    }
+
+    // Remaining args for function
+    PyObject *fArgs = PyTuple_GetSlice(args, 4, numargs);  // New reference
+    if (fArgs == NULL) {
+        return NULL;
+    }
+    Py_INCREF(function);
+
+    callback_data *cb_data = callback_data_new(function, fArgs, NULL);
+    if (cb_data == NULL) {
+        Py_DECREF(function);
+        Py_DECREF(fArgs);
```

Comment:
Rather than handle all the appropriate deallocations, just `DEFER(Py_DECREF(function))` after successfully creating the object.

Reply:
This won't work, these objects need to be moved into the callback later. Disregard.